app.py

```
@@ -102,132 +120,42 @@ def get_frame_range_from_shotgun(self):
-            
-            # set frame ranges for rendering
-            Application.SetValue("Passes.RenderOptions.FrameStart", in_frame)
-            Application.SetValue("Passes.RenderOptions.FrameEnd", out_frame)
-            
-
-        elif engine == "tk-houdini":
-            import hou
-            # We have to use hscript until SideFX gets around to implementing hou.setGlobalFrameRange()
-            hou.hscript("tset `((%s-1)/$FPS)` `(%s/$FPS)`" % (in_frame, out_frame))            
-            hou.playbar.setPlaybackRange(in_frame, out_frame)
-
-        elif engine == "tk-3dsmax":
-            from Py3dsMax import mxs
-            mxs.animationRange = mxs.interval(in_frame, out_frame)
-        elif engine == "tk-3dsmaxplus":
-            import MaxPlus 
-            ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
-            range = MaxPlus.Interval(in_frame * ticks, out_frame * ticks)
-            MaxPlus.Animation.SetRange(range)
-        
-        else:
-            raise tank.TankError("Don't know how to set current frame range for engine %s!" % engine)
+        try:
+            result = self.execute_hook("hook_frame_operation",
+                                       operation="set_frame_range",
+                                       in_frame=in_frame,
+                                       out_frame=out_frame)     
+        except tank.TankError, e:
+            # deliberately filter out exception that used to be thrown 
+            # from the scene operation hook but has since been removed
+            if not str(e).startswith("Not supported frame operation '"):
+                # just re-raise the exception:
+                raise
+        return result
+
+    def get_default_frame_range(self):
+        default_in = self.get_setting("default_sg_in_frame_value")
+        default_out = self.get_setting("default_sg_out_frame_value")
+        return (default_in, default_out)
```

You need exception handling for when these values aren't set. You should raise an Exception and catch above if necessary.

---

If these values are not set, then it will be caught at the time of loading the app. So by the time we reach here, we can safely assume that the values are already set.

Under ___load_apps_ in engine.py:
```
try:
    app_obj = load_application(self, self.context, self.env, app_instance_name)

except TankError as e:
    # validation error - probably some issue with the settings!
    # report this as an error message.
    self.log_error("App configuration Error for %s (configured in environment '%s'). "
                   "It will not be loaded: %s" % (app_instance_name, self.env.disk_location, e))
    continue
```

---

We shouldn't force the definition of default values in the configuration. You should specify `allows_empty: True` in info.yml and then catch it here.

---

Looks like we do not have `allows_empty: True` for `int` type settings. Moving it into a `dict` type setting so as to enable `allows_empty: True`.